### PR TITLE
Servoshell: Add support for mouse thumb buttons to go back/forward

### LINKED ISSUE #31655: servoshell: make the mouse thumb buttons go back and forward
The lower thumb button should go back, and the upper thumb button should go forward. These buttons are sometimes known as “mouse4” and “mouse5”, “X1” and “X2”, or buttons 8 and 9.

The code that handles mouse input is in [Minibrowser::on_event](https://github.com/servo/servo/blob/f3a73dbed39c38c6581fc356d3d4f2d38c906e93/ports/servoshell/minibrowser.rs#L89-L111).

This may be a good [newcomer issue](https://github.com/servo/servo/labels/B-newcomer) once winit is upgraded to 0.29 (#31278).

### PR DESCRIPTION
**Pull request by MunishMummadi**
This patch adds support to make mouse buttons go backward and forward in servoshell's minibrowser.

[Refer to this doc](https://docs.rs/winit/latest/winit/event/enum.MouseButton.html#variant.Back)
-  And this line `macOS: Back and Forward might not work with all hardware.`
- The feature doesn't seem to work on my mac after changes.
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #31655

### CODE REVIEW

File: ports/servoshell/minibrowser.rs [hunk 1]
```diff
@@ -107,10 +108,20 @@ impl Minibrowser {
                 self.last_mouse_position
                     .map_or(false, |p| self.is_in_browser_rect(p))
             },
-            winit::event::WindowEvent::MouseWheel { .. } |
-            winit::event::WindowEvent::MouseInput { .. } => self
-                .last_mouse_position
-                .map_or(false, |p| self.is_in_browser_rect(p)),
```
Comment (mrobinson) [inline ports/servoshell/minibrowser.rs:113]: It looks like this change no longer checks to see if the last mouse position was in the browser rectangle? What's going on here?

**Comment by mrobinson** [comment]
It's a bit unclear from your message: have you verified that this change works or does it just not work on Mac?

**Comment by MunishMummadi** [comment]
I read that it doesn't work on some mac hardware(I might be completely wrong). Even though I removed `MouseWheel` event , the mouse buttons should work right?. But removing both `MouseWheel` & `MouseInput` isn't a good idea. I will change it.

File: ports/servoshell/minibrowser.rs (hunk 1 above)
Comment (MunishMummadi) [reply ports/servoshell/minibrowser.rs:113]: Yeah, I thought I made a way around i.e I simplified things. But it isn't the case. I will try to do some changes.

**Comment by MunishMummadi** [comment]
Even though `./mach build --dev` doesn't produce any errors. I am not able to create the functionality. So, I am making this PR a draft.

**Comment by mrobinson** [comment]
If this isn't functional at all, let's close this one for now to clear it out of the queue. Once you have this working just open it up again and I'll be more than happy with to take a look at it. Thanks!

**Comment by MunishMummadi** [comment]
Got it!!

**Pull request closed without merging**